packages/block-editor/src/store/private-actions.js

```diff
@@ -140,6 +140,15 @@ export const privateRemoveBlocks =
 				}
 
 				if ( rules[ 'bindings/core/pattern-overrides' ] ) {
+					const parentPatternBlocks =
+						select.getBlockParentsByBlockName(
+							clientId,
+							'core/block'
+						);
+					// We only need to run this check when editing the original pattern, not pattern instances.
+					if ( parentPatternBlocks?.length > 0 ) {
+						continue;
+					}
```

Comment:
I guess another way is to only have the pattern overrides rule when within a pattern.

No real preference here, but I think it might affect the follow up that's underway.